e2e_playwright/multipage_apps_v2/mpa_v2_basics.py

```diff
@@ -0,0 +1,95 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pathlib
+
+from PIL import Image
+
+import streamlit as st
+
+parent_directory = pathlib.Path(__file__).parent.parent / "multipage_apps"
+small_logo = Image.open(str(parent_directory / "small-streamlit.png"))
+
+logo = Image.open(str(parent_directory / "full-streamlit.png"))
+
+st.logo(logo, link="https://www.example.com", icon_image=small_logo)
+
+st.header("Main Page")
+x = st.slider("x")
+
+st.write(f"x is {x}")
+
+set_default = bool(st.query_params.get("default", False))
+
+page2 = st.Page("page_2.py")
+page3 = st.Page("page_3.py", title="Different Title")
+page4 = st.Page("🦒_page_4.py")
+page5 = st.Page("page_5.py", icon=":material/settings:")
+page6 = st.Page("page_6.py", default=set_default)
+
+
+def page_7():
+    st.header("Page 7")
+    x = st.slider("y")
+    st.write(f"y is {x}")
+
+
+def page_8():
+    st.header("Page 8")
```

Comment:
Just to make sure I understand. Are you saying the st calls are in `my_app` or in `my_nice_page` as a function? I imagine the `my_nice_page` as a function is not a problem because the function will always be executed. If you are saying in `my_app`. I can test it, but I would imagine it would follow the same trend of the main script. I can try both, but I want to understand the concern.

Reply:
No concern, just a question... Nothing to do here. I will just try it out after the release